data/scripts/lib/register_monster_type.lua

```diff
@@ -247,8 +244,12 @@ registerMonsterType.attacks = function(mtype, mask)
 			if attack.name then
 				if attack.name == "melee" then
 					spell:setType("melee")
-					if attack.attack and attack.skill then
-						spell:setAttackValue(attack.attack, attack.skill)
+					if attack.attack and attack.skill then						
```

Comment:
Does this check make sense? I'm thinking just the contents of it already do the job. Also, you aren't checking if `attack.minDamage` or `attack.maxDamage` are nil too, so why check for `attack.attack` and `attack.skill`?

Reply:
they can be nil in some cases, more than anything so that when converting xml monsters to lua, they do not suffer from a little known problem, which causes them not to have the same damage that was configured in xml, if no one else provides any proof that this may be of help, then I'll close it in a moment.